File: non-native/zinotes_flutter/lib/model/hanzi.dart
class Hanzi{
  final String id;
  final String pinyin;
  final List<int> tones;
  final int? radicalNumber;
  final int? strokeCount;
  final int? hskLevel;
  final List<String>? definitions;

  Hanzi({required this.id, required this.pinyin, required this.tones, this.radicalNumber, this.strokeCount, this.hskLevel, this.definitions});
}

File: native/ZiNotes/app/src/main/java/com/example/zinotes/room/Hanzi.kt
package com.example.zinotes.room

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

@Serializable
@Entity(tableName = "hanzi")
data class Hanzi (
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,

    val pinyin: String = "",

    val tones: List<Int> = listOf(),

    val radicalNumber: Int? = null,

    val strokeCount: Int? = null,

    val hskLevel: Int? = null,

    val definitions: List<String>? = null,

    @Transient
    val isSynced: Boolean = true,

    @Transient
    val isDeleted: Boolean = false,

    @Transient
    val isNew: Boolean = false
)